src/govuk/common/index.mjs

```diff
@@ -162,9 +163,10 @@ export function extractConfigByNamespace (configObject, namespace) {
 }
 
 /**
+ * @template {Node} ElementType
```

Comment:
The same Generics system as for `nodeListForEach`allows to define iterators expecting specific kinds of elements as their first parameter (as well as the `NodeList` they receive as last parameter). For example,  a `nodeListIterator<HTMLInputElement>` that's meant to iterate on `<input>` elements.

That allows TypeScript to keep the trail of types when using `nodeListForEach`, and receive methods meant to iterate on sub-classes of `Element`. That's something it couldn't do before, as TypeScript would see `nodeListForEach` trying to run an iterator expecting `HTMLInputElement`, but could only acertain that it would pass it an `Element`, so bring that to our attention (and force us to use a wider type for the iterator and an `instanceof` to narrow down within the function).

Reply:
I wasn't aware it had to be the first line. Just put it there as it made sense to announce first that the block was using generics then use generics 😄